dask/tests/test_base.py

```diff
@@ -954,6 +954,17 @@ def test_visualize():
         visualize(x, filename=os.path.join(d, "mydask.png"))
         assert os.path.exists(os.path.join(d, "mydask.png"))
 
+        test_list = []
```

Comment:
It's confusing to work out what is actually being tested just from reading the new test code you've added. To address this, I'd probably recommend splitting it up into it's own separate test and call that something like `test_visualize_no_filename`. Then the name of the test can do the work of describing what the test is for.

Whether something should be a new test, or part of an existing test is a bit of a borderline judgement call, and other people might disagree with me. Ultimately, I think it'lll save time for future developers because it's a clearer way to mark "you can ignore this bit" if they're not looking into the specific case where `filename=None`.

Reply:
Yes, this seems to be a better option.

```py
@pytest.mark.skipif("not da")
@pytest.mark.skipif(
    sys.flags.optimize, reason="graphviz exception with Python -OO flag"
)
def test_visualize_no_filename(monkeypatch):
    pytest.importorskip("graphviz")
    pytest.importorskip("dask.dot")

    x = da.arange(10)

    test_list = []
    _mock_get_display_cls = dask.dot._get_display_cls

    def mock_fx(format):
        test_list.append(True)
        return _mock_get_display_cls(format)

    monkeypatch.setattr(dask.dot, "_get_display_cls", mock_fx)
    visualize(x, filename=None)
    assert True in test_list
```
  
I can add this (the test :heavy_check_mark:)
I will wait for the final decision on using monkeypatch to mock behavior and then commit accordingly!

Reply:
I have no opinion on one test versus two